doped/utils/parsing.py

```
@@ -1988,3 +2009,374 @@ def _multiple_files_warning(file_type, directory, chosen_filepath, action=None,
+        """
+        Compute NELECT for a QE system using a Vasprun object and QE UPF pseudopotentials.
+
+        Args:
+            vasprun (Vasprun): VASP vasprun.xml parsed object.
+            pp_folder (str | Path): Path to QE UPF pseudopotential folder.
+
+        Returns:
+            float: Total number of electrons for the system (neutral).
+        """
+        def _get_zval_from_upf(pp_file: Path) -> float:
+            """Extract z_valence from a QE UPF pseudopotential file (supports v1, v2 XML, and old format)."""
```

Ruff (0.12.2)

Line too long (109 > 107)

(E501)